### PR TITLE
fix: nodeSelector not applied to kratos courier

### PR DESCRIPTION
**Pull request by alexGNX**
This is a fix to apply the nodeSelector that is applied to kratos deployment to courier statefulSet

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
      and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added necessary documentation within the code base (if
      appropriate).

### CODE REVIEW

File: helm/charts/kratos/Chart.yaml [hunk 1]
```diff
@@ -3,5 +3,5 @@ appVersion: v0.9.0-alpha.2
 description: A ORY Kratos Helm chart for Kubernetes
 name: kratos
 icon: https://raw.githubusercontent.com/ory/docs/master/docs/static/img/logo-kratos.svg
-version: 0.23.2
+version: 0.23.3
```
Comment (Demonsthere) [inline helm/charts/kratos/Chart.yaml:6]: This is auto-generated during release, please do not modify it manually 😉

File: helm/charts/kratos/templates/statefulset-mail.yaml [hunk 2]
```diff
@@ -133,3 +133,7 @@ spec:
             name: {{ include "kratos.fullname" $root }}-template-{{ $method }}-{{ $result }}
         {{- end }}
         {{- end }}
+      {{- with .Values.nodeSelector }}
```
Comment (Demonsthere) [inline helm/charts/kratos/templates/statefulset-mail.yaml:136]: Please add a ternary operation to the selector like with 
`{{- $extraVolumes := ternary .Values.statefulSet.extraVolumes .Values.deployment.extraVolumes (not (empty .Values.statefulSet.extraVolumes )) -}}`
To allow setting it for the statefulset and deployment separately

**Review by Demonsthere — CHANGES_REQUESTED** [review]

File: helm/charts/kratos/Chart.yaml (hunk 1 above)
Comment (alexGNX) [reply helm/charts/kratos/Chart.yaml]: oh that's right. Done.

**Review by Demonsthere — APPROVED** [review]
Thanks! 😃

File: helm/charts/kratos/templates/statefulset-mail.yaml (hunk 2 above)
Comment (alexGNX) [reply helm/charts/kratos/templates/statefulset-mail.yaml]: ok i see, issue is there is a `.Values.nodeSelector` and a `.Values.deployment.nodeSelector`, and only the `.Values.nodeSelector` was used. I did something, tell me what you think about it.

**Pull request merged**